_posts/2024-12-19-cosmicqc.md

```diff
@@ -0,0 +1,313 @@
+  width="100%"
+%}
+> Logo for coSMicQC being shown on a stage, like the star of the show! ⭐️
+
+I want to shout out Dave Bunten (aka @d33bs) here, who has been a major contributor to coSMicQC, which is an open-source software that we are developing with sustainable software development practices! 📣
+
+In summary, coSMicQC uses extracted morphology features directly from CellProfiler, performs normalization of specific features that are most related to detecting poor segmentation, and uses standard deviation to identify single cells that are considered technical outliers.
+
+Going into more detail on this new method, it works by first creating conditions.
+A condition is a feature or group of features that will capture a specific type of technical artifact.
+For example, nuclei area and intensity can work together as a condition to detect large nuclei with high intensity which represents segmentation of clustered/overlapping nuclei.
+Your condition will also need how many standard deviations from the mean that you expect these outliers to be located.
+If you want to detect cells above a mean, you use a positive value, and vice versa for cells below the mean.
+
+{%
+  include figure.html
+  image="images/blog/cosmicqc/area_outlier_example.png"
+  width="100%"
+%}
+> Example of how outliers are detected based on nuclei area in coSMicQC.
+
+You can use the function `find_outliers` to the simple conditions and voila, coSMicQC yields a dataframe with flagged single-cell outliers per condition.
+A summary is also outputted, which includes the total number of outliers detected, the percent of the total cells in the data detected as outliers, and the range of the values of the outliers for the features used in the condition.
+
+A highlight of coSMicQC is that you are able to visualize single-cell segmentations within the notebook if you provide the original images and exported outlines from CellProfiler, through the use of a novel dataframe format, built on top of `pandas`, that we call [CytoDataFrame](https://github.com/WayScience/CytoDataFrame).
+This open-source formatter is also developed by Dave Bunten! 📣
+This makes it more streamlined to assess if your conditions are working without having to go outside of the notebook and jump to your handy file manager, Fiji, or CellProfiler to see what the segmented cells look like.
+This feature is still being developed as we find more datasets with varying conditions for image file structure.
+
+Once you have optimized the conditions, you can then use the function `label_outliers` which will output the original dataframe with columns for each condition and if a single-cell passed or failed each condition.
+You can then choose to either save the original dataframe with the new columns and keep outliers or remove single cells that were detected as outliers.
+You can also not use the coSMicQC functionality and take the indices from the `find_outliers` outputs and remove single cells that way.
```

Comment:
consider deleting this

```suggestion
```

Reply:
Fair, I think the main point is detecting the outliers, people can then do what they want. I might keep 219 though maybe.